File: backend/app/schemas/investor.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class InvestorBase(BaseModel):
    code: str
    name: str
    role: str = "viewer"
    phone: Optional[str] = None
    email: Optional[str] = None


class InvestorCreate(InvestorBase):
    password: str


class InvestorUpdate(BaseModel):
    name: Optional[str] = None
    role: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None


class InvestorResponse(InvestorBase):
    last_login_at: Optional[datetime] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: frontend/src/types/investor.ts
export interface Investor {
  code: string;
  name: string;
  role: string;
  phone?: string;
  email?: string;
  last_login_at?: string;
  created_at?: string;
  updated_at?: string;
}

export interface InvestorCreate {
  code: string;
  name: string;
  role?: string;
  phone?: string;
  email?: string;
  password: string;
}

export interface InvestorUpdate {
  name?: string;
  role?: string;
  phone?: string;
  email?: string;
}
